test/numeric-separators-style.js

```diff
@@ -0,0 +1,308 @@
+		},
+		{
+			code: 'var foo = 0b10101010101010',
+			options: [{binary: {minimumThreshold: 0, preferedGroupLength: 4}}],
+			errors: [error],
+			output: 'var foo = 0b10_1010_1010_1010'
+		},
+		{
+			code: 'var foo = 1_9_223n',
+			options: [{bigint: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 19_223n'
+		},
+		{
+			code: 'var foo = 80_7n',
+			options: [{bigint: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 807n'
+		},
+		{
+			code: 'var foo = 123456789_100n',
+			options: [{bigint: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 123_456_789_100n'
+		},
+		{
+			code: 'var foo = 1_2_345_678',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 12_345_678'
+		},
+		{
+			code: 'var foo = 12_3',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 123'
+		},
+		{
+			code: 'var foo = 1234567890',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 1_234_567_890'
+		},
+		{
+			code: 'var foo = 9807.1234567',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 9807.123_456_7'
+		},
+		{
+			code: 'var foo = 3819.123_4325',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 3819.123_432_5'
+		},
+		{
+			code: 'var foo = 138789.12343_2_42',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 138_789.123_432_42'
+		},
+		{
+			code: 'var foo = .000000_1',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = .000_000_1'
+		},
+		{
+			code: 'var foo = -100000_1',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = -1_000_001'
+		},
+		{
+			code: 'var foo = 1e10000',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 1e10_000'
+		},
+		{
+			code: 'var foo = 39804e10000',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 39_804e10_000'
+		},
+		{
+			code: 'var foo = -123456e100',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = -123_456e100'
+		},
+		{
+			code: 'var foo = 3.6e12000',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = 3.6e12_000'
+		},
+		{
+			code: 'var foo = -1200000e5',
+			options: [{number: {minimumThreshold: 5, preferedGroupLength: 3}}],
+			errors: [error],
+			output: 'var foo = -1_200_000e5'
+		}
+	]
+});
```

Comment:
Legacy `0777777` need test.

Reply:
How should it work then? Is the leading zero considered as a "regular" digit, and should be correctly grouped, or is it considered as a "prefix" and ignores grouping?
Maybe it can even be fixed to `0o`, as `0` is now deprecated? Although this might be a breaking change, users of eslint and eslint-plugin-unicorn should have at least node 10, and afaik, `0o` is supported in node 10... (need verification though)

Reply:
Just make sure  we don't break it, like this  https://github.com/sindresorhus/eslint-plugin-unicorn/blob/f141d08829085cb314bc78a4bdf59e2c739dedb8/test/number-literal-case.js#L25

Reply:
Just make sure  we don't break it, like this  https://github.com/sindresorhus/eslint-plugin-unicorn/blob/f141d08829085cb314bc78a4bdf59e2c739dedb8/test/number-literal-case.js#L25